control-plane/api/mesh/v2beta1/gateway_class_config_types.go

```diff
@@ -73,7 +73,8 @@ type GatewayClassDeploymentConfig struct {
 	// More info: https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/
 	TopologySpreadConstraints []corev1.TopologySpreadConstraint `json:"topologySpreadConstraints,omitempty"`
 	// DNSPolicy specifies the dns policy to use. These are set on a per pod basis.
-	DNSPolicy string `json:"dnsPolicy,omitempty"`
+	// +kubebuilder:validation:Enum=Default;ClusterFirst;ClusterFirstWithHostNet;None
+	DNSPolicy corev1.DNSPolicy `json:"dnsPolicy,omitempty"`
```

Comment:
Is it safe to assume that this validation only applies if the key is included in the yaml? So the fact that it would obviously default to `""` if not set would be fine?

Reply:
yup just tested it, this only applies if the value is specified (and k8s defaults to using `ClusterFirst` here if the value is not set https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy)